### PR TITLE
Removal of Spout Class Referecneces Attempt 2.

### PR DESCRIPTION
**Pull request by Waterpicker**
...nt side works

Waterpicker Waterpicker@rocketmail.com

### CODE REVIEW

**Comment by Sleaker** [comment]
I don't like just blindly passing around the VanillaPlugin instance all over the place. Also not a a fan of the static access for the filesystem reference, depending on how classloading works it's liable to produce unknown behaviour (ie break)

I think a few of these classes need to be changed so they aren't just static lists, they need to be instantiated, or passed in a reference to the engine (recipe manager etc).  

I'm not sure if this can be pulled, though it looks like it's mostly fine.

**Comment by Waterpicker** [comment]
Yea. The purpose of this pull is not meant to be a final edit. Just enough so Affy can go ahead with his work on launching and stuff.

File: src/main/java/org/spout/vanilla/VanillaListener.java [hunk 1]
```diff
@@ -102,11 +101,11 @@ public void onPlayerJoin(PlayerJoinEvent event) {
 
 	@EventHandler
 	public void onGameStart(EngineStartEvent event) {
-		if (Spout.getPlatform() != Platform.CLIENT) {
+		if (VanillaPlugin.getInstance().getEngine().getPlatform() != Platform.CLIENT) {
```
Comment (Afforess) [inline src/main/java/org/spout/vanilla/VanillaListener.java:104]: Event should contain engine instance

File: src/main/java/org/spout/vanilla/VanillaListener.java [hunk 2]
```diff
@@ -164,7 +163,7 @@ public void onBlockChange(BlockChangeEvent event) {
 		}
 		if (prevPower != -1) {
 			RedstoneChangeEvent redstoneEvent = new RedstoneChangeEvent(event.getBlock(), event.getCause(), prevPower, newPower);
-			Spout.getEventManager().callEvent(redstoneEvent);
+			VanillaPlugin.getInstance().getEngine().getEventManager().callEvent(redstoneEvent);
```
Comment (Afforess) [inline src/main/java/org/spout/vanilla/VanillaListener.java:166]: Doesn't the listener get passed the engine instance when it is created? If not, it should.

File: src/main/java/org/spout/vanilla/component/entity/substance/test/SceneOutput.java [hunk 3]
```diff
@@ -26,15 +26,16 @@
...
-import org.spout.api.Spout;
+
 import org.spout.api.component.type.EntityComponent;
 import org.spout.api.entity.Player;
 
 public class SceneOutput extends EntityComponent {
 	@Override
 	public void onAttached() {
-		if (Spout.getPlatform() != Platform.SERVER || !Spout.debugMode() || getOwner() instanceof Player) {
+		if (getOwner().getEngine().getPlatform() != Platform.SERVER || !getOwner().getEngine().debugMode() || getOwner() instanceof Player) {
```
Comment (Afforess) [inline src/main/java/org/spout/vanilla/component/entity/substance/test/SceneOutput.java:38]: No need for getOwner().getEngine(), you can do just getEngine()

**Comment by windy1** [comment]
Do we actually _need_ `VanillaPlugin.getInstance()` for anything besides convenience? If not, I think we should remove it anyways.

**Comment by Waterpicker** [comment]
I only used what I had at my disposal.

**Comment by Zidane** [comment]
@W1ndwaker 

Sadly Materials need it for initialize in VanillaMaterials else we would have already removed it.

**Comment by Afforess** [comment]
Merged.

**Pull request merged**